### ISSUE TITLE
@torch.jit.script causes compilation error on Ampere architecture (RTX 3090)

### ISSUE THREAD

**Issue by Callidior** [pytorch/pytorch#48200]
## 🐛 Bug

For being able to use a RTX 3090, I upgraded my PyTorch to 1.7.0 using conda. When wanting to run my code, which used to run perfectly on PyTorch 1.6 and GTX 2080 Ti, I encountered the following error. The stack trace mentions a function that uses `@torch.jit.script`. When I remove this decorator, the code runs fine.

```
Traceback (most recent call last):
  File "example.py", line 12, in <module>
    out = test(A).cpu().item()
RuntimeError: nvrtc: error: invalid value for --gpu-architecture (-arch)

nvrtc compilation failed:

#define NAN __int_as_float(0x7fffffff)
#define POS_INFINITY __int_as_float(0x7f800000)
#define NEG_INFINITY __int_as_float(0xff800000)


template<typename T>
__device__ T maximum(T a, T b) {
  return isnan(a) ? a : (a > b ? a : b);
}

template<typename T>
__device__ T minimum(T a, T b) {
  return isnan(a) ? a : (a < b ? a : b);
}

extern "C" __global__
void func_1(float* t0, float* t1, float* aten_sub_flat) {
{
  if (512 * blockIdx.x + threadIdx.x<1000 ? 1 : 0) {
    float v = __ldg(t0 + (((512 * blockIdx.x + threadIdx.x) / 10) / 10) * 10 + (512 * blockIdx.x + threadIdx.x) % 10);
    float v_1 = __ldg(t1 + (512 * blockIdx.x + threadIdx.x) / 10);
    aten_sub_flat[512 * blockIdx.x + threadIdx.x] = v - v_1;
  }
}
}
```

The same code still runs perfectly with PyTorch 1.7 on a GTX 2080, even with the decorator.

## To Reproduce

Run the following code on an RTX 3090 (or other Ampere GPU, probably):

```python
import torch

@torch.jit.script
def test(A):

    D = A[:, None, :] - A[:, :, None]
    return torch.mean(D)

if __name__ == '__main__':
    for i in range(2):
        A = torch.rand(10, 10).cuda()
        out = test(A).cpu().item()
        print(out)
```

Note that the first call to `test` is successful. It fails on the second one.

After removing `@torch.jit.script`, the error should disappear.

## Expected behavior

The function is being executed on the GPU without error and two numbers are printed.

## Environment

```
PyTorch version: 1.7.0
Is debug build: True
CUDA used to build PyTorch: 11.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.5 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: Could not collect

Python version: 3.8 (64-bit runtime)
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration:
GPU 0: GeForce RTX 3090
GPU 1: GeForce RTX 2080 Ti

Nvidia driver version: 455.38
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.19.2
[pip3] torch==1.7.0
[pip3] torchaudio==0.7.0a0+ac17b64
[pip3] torchtext==0.8.0a0+cd6902d
[pip3] torchvision==0.8.1
[conda] blas                      1.0                         mkl
[conda] cudatoolkit               11.0.221             h6bb024c_0
[conda] mkl                       2020.2                      256
[conda] mkl-service               2.3.0            py38he904b0f_0
[conda] mkl_fft                   1.2.0            py38h23d657b_0
[conda] mkl_random                1.1.1            py38h0573a6f_0
[conda] numpy                     1.19.2           py38h54aff64_0
[conda] numpy-base                1.19.2           py38hfa32c7d_0
[conda] pytorch                   1.7.0           py3.8_cuda11.0.221_cudnn8.0.3_0    pytorch
[conda] torchaudio                0.7.0                      py38    pytorch
[conda] torchtext                 0.8.0                      py38    pytorch
[conda] torchvision               0.8.1                py38_cu110    pytorch
```

cc @gmagogsfm

**Comment by Callidior** [comment]
Okay, this seems to not necessarily be a PyTorch bug, I realise. A look at the documentation of `nvrtc` included with [CUDA Toolkit 11.0](https://docs.nvidia.com/cuda/archive/11.0/nvrtc/#group__options) reveals that it indeed does not support the Ampere architecture (`compute_80`). This support is introduced in [CUDA Toolkit 11.1](https://docs.nvidia.com/cuda/archive/11.1.0/nvrtc/#group__options).

Unfortunately, the conda repositories do not contain builds of PyTorch against CUDA 11.1.

**Comment by jjsjann123** [comment]
CUDA 11.0 supports sm_80 but nvrtc shipped with it does not... I can try to clamp it on the right side to compute_75. :crossed_fingers:

**Comment by jjsjann123** [comment]
Just to correct some information regarding docs. nvrtc in CUDA  11.0 does support `compute_80`. The documentation made an unfortunate mistake (out of sync). 3090 has `compute_86`.
This issue should have been fixed with #48151 already.
